app/scripts/models/auth_brokers/oauth.js

```diff
@@ -102,6 +118,36 @@ define(function (require, exports, module) {
         .then(_formatOAuthResult);
     },
 
+    /**
+     * Derive scoped keys and encrypt them with the relier's public JWK
+     *
+     * @param {Object} account
+     * @param {String} assertion
+     * @returns {Promise} Returns a promise that resolves into an encrypted bundle
+     * @private
+     */
+    _provisionScopedKeys (account, assertion) {
+      const relier = this.relier;
+      const keyFetchToken = account.get('keyFetchToken');
+      const unwrapBKey = account.get('unwrapBKey');
+
+      // if check if requested scopes provide scoped keys
+      return this._oAuthClient.getClientKeyData({
+        assertion: assertion,
+        client_id: relier.get('clientId'), //eslint-disable-line camelcase
+        scope: decodeURIComponent(relier.get('scope'))
+      }).then((clientKeyData) => {
+        if (! unwrapBKey || ! keyFetchToken || Object.keys(clientKeyData).length === 0) {
+          // if we got no keys or key data then exit out
+          return null;
+        }
+
```

Comment:
The account should already have a reference to `fxaClient` and `oAuthClient`, can fetching the account keys be done over there instead? It seems like deriving keys for an account/relier pair is an account related activity already, and by doing so, the broker won't need its own references to `fxaClient` and `oAuthClient`